components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -0,0 +1,190 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBannerView.h"
+
+#import "MDCBannerParams.h"
+#import "MDCBannerViewLayout.h"
+#import "MaterialColorScheme.h"
+
+@interface MDCBannerView () <MDCBannerViewLayoutDataSource>
+
+@property(nonatomic, readwrite, weak) UILabel *textLabel;
+@property(nonatomic, readwrite, weak) UIView *iconImageViewContainer;
+@property(nonatomic, readwrite, weak) UIImageView *iconImageView;
+
+@property(nonatomic, readwrite, strong) MDCBannerViewLayout *layout;
+
+@end
+
+@implementation MDCBannerView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonBannerViewInit];
+  }
+  return self;
+}
+
+- (void)commonBannerViewInit {
+  self.backgroundColor = [UIColor whiteColor];
+  _buttons = [[NSMutableArray alloc] init];
+}
+
+#pragma mark - Property Setter and Getter
+
+- (void)setText:(NSString *)text {
+  if (text) {
+    _text = text;
+    UILabel *textLabel = self.textLabel;
+    if (!textLabel) {
+      textLabel = [[UILabel alloc] init];
```

Comment:
Please move any subview initialization logic to the view's commonBannerViewInit method. Lazy creation doesn't save us much for this component in terms of performance or memory, but it does increase the local complexity of these setter methods.

Reply:
Addressed in 078cff215c7c83a3d5899a1f3cfa53a0343149b9